Looks like there's a golint issue;

```
09:39:27 Errors from golint:
09:39:27 daemon/daemon.go:766:1: comment on exported method Daemon.WaitToWork should be of the form "WaitToWork ..."
09:39:27 
09:39:27 Please fix the above errors. You can test via "golint" and commit the result.
09:39:27 
```

--- a/daemon/daemon.go
+++ b/daemon/daemon.go
@@ -116,6 +116,7 @@ type Daemon struct {
 	diskUsageRunning int32
 	pruneRunning     int32
 	hosts            map[string]bool // hosts stores the addresses the daemon is listening on
+	startupDone      chan struct{}
 }
 
 // StoreHosts stores the addresses the daemon is listening on
@@ -543,7 +544,10 @@ func NewDaemon(config *config.Config, registryService registry.Service, containe
 	}
 	os.Setenv("TMPDIR", realTmp)
 
-	d := &Daemon{configStore: config}
+	d := &Daemon{
+		configStore: config,
+		startupDone: make(chan struct{}),
+	}
 	// Ensure the daemon is properly shutdown if there is a failure during
 	// initialization
 	defer func() {
@@ -740,6 +744,7 @@ func NewDaemon(config *config.Config, registryService registry.Service, containe
 	if err := d.restore(); err != nil {
 		return nil, err
 	}
+	close(d.startupDone)
 
 	// FIXME: this method never returns an error
 	info, _ := d.SystemInfo()
@@ -758,6 +763,10 @@ func NewDaemon(config *config.Config, registryService registry.Service, containe
 	engineMemory.Set(float64(info.MemTotal))
 
 	return d, nil
+}
+
+func (daemon *Daemon) waitForStartupDone() {
+	<-daemon.startupDone
 }
 
 func (daemon *Daemon) shutdownContainer(c *container.Container) error {

--- a/daemon/monitor.go
+++ b/daemon/monitor.go
@@ -69,6 +69,10 @@ func (daemon *Daemon) StateChanged(id string, e libcontainerd.StateInfo) error {
 			go func() {
 				err := <-wait
 				if err == nil {
+					// daemon.netController is initialized when daemon is restoring containers.
+					// But containerStart will use daemon.netController segment.
+					// So to avoid panic at startup process, here must wait util daemon restore done.
+					daemon.waitForStartupDone()
 					if err = daemon.containerStart(c, "", "", false); err != nil {
 						logrus.Debugf("failed to restart container: %+v", err)
 					}

